cityofaustin/atd-vz-data
# Add afd__incidents table schema and trigger to LDM

## Associated issues
https://github.com/cityofaustin/atd-data-tech/issues/13698
https://github.com/cityofaustin/atd-data-tech/issues/14140

This PR adds migrations for the `afd__incidents` table schema (no data) and the one trigger set on it. I'm using the same patterns that Frank started in https://github.com/cityofaustin/atd-vz-data/pull/1305, and the idea is to apply these migrations and backfill the data on LDM staging when this code merges. We'll keep building that instance up while building up the LDM transformation steps (transform script + migrations + backfills).

## Testing
**URL to test:** 
Local

**Steps to test:**
1. Make sure that your `.env` is updated with the `LDM_` variables shown in the `env_template`. Check the 1Pass secrets called `RDS Instance: LDM staging` for the values. `LDM_DATABASE` is `atd_vz_data`.
2. Replicate the LDM staging database locally by running:
```bash
$ ./vision-zero -l replicate-db
```
3. From the `atd-vzd` folder run:
```bash
$ hasura migrate apply --envfile .env.local
$ hasura metadata apply --envfile .env.local
```
4. To backfill the EMS incidents data from the VZ read replica to your local LDM database, run:
```bash
export RR_PASSWORD=<your VZ read replica password>
export RR_USER=<your VZ read replica username>

docker exec -i \
-e PGHOST=db-rr.vision-zero.austinmobility.io \
-e PGDATABASE=atd_vz_data \
-e PGPASSWORD=$RR_PASSWORD \
-e PGUSER=$RR_USER \
visionzero-postgis \
pg_dump -t afd__incidents --data-only | \
perl -pe 's/^(SELECT pg_catalog\.set_config)/-- $1/g' | \
docker exec -i \
-e PGHOST=localhost \
-e PGDATABASE=atd_vz_data \
-e PGPASSWORD=visionzero \
-e PGUSER=visionzero \
visionzero-postgis \
psql --set ON_ERROR_STOP=on;
``` 
5. Through your database explorer or the local Hasura console, look for the `public.afd__incidents` table
6. You should see the columns set in the `public.afd_incidents_trigger()` function populated with data (if there are values to generate from) which happened on insert.

---
#### Ship list
